internal/dag/policy.go

```diff
@@ -25,15 +26,28 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
+func retryOn(ro []projcontour.RetryOn) string {
+	if len(ro) == 0 {
+		return "5xx"
+	}
+
+	ss := make([]string, len(ro))
+	for i, value := range ro {
+		ss[i] = string(value)
+	}
+	return strings.Join(ss, ",")
+}
+
 func retryPolicy(rp *projcontour.RetryPolicy) *RetryPolicy {
 	if rp == nil {
 		return nil
 	}
 	perTryTimeout, _ := time.ParseDuration(rp.PerTryTimeout)
 	return &RetryPolicy{
-		RetryOn:       "5xx",
-		NumRetries:    max(1, uint32(rp.NumRetries)),
-		PerTryTimeout: perTryTimeout,
+		RetryOn:              retryOn(rp.RetryOn),
+		RetriableStatusCodes: rp.RetriableStatusCodes,
```

Comment:
What does Envoy do if you send it status codes that are out of range. e.g. 9000?

Reply:
Looks like Envoy will accept any `uint32` as a retriable status code:
https://github.com/envoyproxy/envoy/blob/master/source/common/router/config_impl.cc#L99-L101
And if the response status code matches any retriable status code, regardless of its validity, it will retry it:
https://github.com/envoyproxy/envoy/blob/master/source/common/router/retry_state_impl.cc#L307-L313

If you'd like I can verify this behavior by spinning up Envoy and sending it a payload to see how it behaves.